base/deprecated.jl

```diff
@@ -1179,6 +1181,65 @@ end
 isequal(x::Char, y::Integer) = false
 isequal(x::Integer, y::Char) = false
 
+#6674 and #4233
+macro windows(qm,ex)
+    depwarn("`@windows` is deprecated, use `@static is_windows()` instead", Symbol("@windows"))
+    return @static is_windows() ? esc(ex.args[1]) : esc(ex.args[2])
+end
+macro unix(qm,ex)
+    depwarn("`@unix` is deprecated, use `@static is_unix()` instead", Symbol("@unix"))
+    return @static is_unix() ? esc(ex.args[1]) : esc(ex.args[2])
+end
+macro osx(qm,ex)
+    depwarn("`@osx` is deprecated, use `@static is_apple()` instead", Symbol("@osx"))
+    return @static is_apple() ? esc(ex.args[1]) : esc(ex.args[2])
+end
+macro linux(qm,ex)
+    depwarn("`@linux` is deprecated, use `@static is_linux()` instead", Symbol("@linux"))
+    return @static is_linux() ? esc(ex.args[1]) : esc(ex.args[2])
+end
+macro windows_only(ex)
+    depwarn("`@windows_only` is deprecated, use `@static is_windows()` instead", Symbol("@windows_only"))
+    return @static if is_windows() esc(ex) end
+end
+macro unix_only(ex)
+    depwarn("`@unix_only` is deprecated, use `@static is_unix()` instead", Symbol("@unix_only"))
+    return @static if is_unix() esc(ex) end
+end
+macro osx_only(ex)
+    depwarn("`@osx_only` is deprecated, use `@static is_apple()` instead", Symbol("@osx_only"))
+    return @static if is_apple() esc(ex) end
+end
+macro linux_only(ex)
+    depwarn("`@linux_only` is deprecated, use `@static is_linux()` instead", Symbol("@linux_only"))
+    return @static if is_linux() esc(ex) end
+end
+export
+    @windows,
+    @unix,
+    @osx,
+    @linux,
+    @windows_only,
+    @unix_only,
+    @osx_only,
+    @linux_only
+
+const OS_NAME =
+    if Sys.KERNEL === :Darwin
+        :OSX
+    elseif Sys.KERNEL === :NT
+        :Windows
+    else
+        Sys.KERNEL
+    end
+deprecate(:OS_NAME) # use Sys.KERNEL now
+
+import .Sys.CPU_CORES
+export CPU_CORES # use the export from Sys (can't deprecate this because it instead deprecates Sys.CPU_CORES)
+
+export WORD_SIZE # use the export from Sys (can't deprecate this here since it's used in base)
```

Comment:
This will be risky and breaking to have in this file where it's intended to be deleted. We should do this elsewhere until we have a proper mechanism for "exported version is deprecated, module qualified version is not"